File: app/webapp/src/app/shared/models/article/article.model.ts
import {AuthorModel} from './author.model';

/**
 * Represent an article.
 */
export interface ArticleModel {
  id?: number;
  title?: string;
  description?: string;
  content?: string;
  isArchived?: boolean;
  author?: AuthorModel;
  authorId?: string;
  createdAt?: Date;
  lastUpdateAt?: Date;
  deletedAt?: Date;
  categories?: Array<number>;
}

File: app/api/src/main/java/fr/koi/wikiapi/web/model/graphql/article/ArticleModel.java
package fr.koi.wikiapi.web.model.graphql.article;

import fr.koi.wikiapi.web.model.graphql.article_category.ArticleCategoryModel;
import fr.koi.wikiapi.web.model.graphql.user.AuthorModel;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

import java.time.ZonedDateTime;
import java.util.List;

@Getter
@Setter
@Accessors(chain = true)
public class ArticleModel {
    private Long id;
    private String title;
    private String description;
    private String content;
    private Boolean isArchived;
    private AuthorModel author;
    private List<ArticleCategoryModel> categories;
    private ZonedDateTime createdAt;
    private ZonedDateTime lastUpdateAt;
    private ZonedDateTime deletedAt;
}
